addon/-private/system/diff-array.js

```diff
@@ -57,3 +58,95 @@ export default function diffArray(oldArray, newArray) {
     removedCount
   };
 }
+
+export function setForArray(array) {
+  let set = new OrderedSet();
+
+  if (array) {
+    for (let i=0, l=array.length; i<l; i++) {
+      set.add(array[i]);
+    }
+  }
+
+  return set;
+}
+
+export function computeChanges(oldArray, newArray, removalSet) {
+  let changeset = {
+    changes: false,
+    additions: [],
+    removals: []
+  };
+
+  let diff = diffArray(oldArray, newArray);
+
+  if (diff.firstChangeIndex === null) {
+    // no changes found
+    if (removalSet) {
+      // records that have been removed since last compute will need their inverses to be corrected
+      for (let i = 0, l = newArray.length; i < l; i++) {
+        let record = newArray[i];
+        if (removalSet.has(record)) {
+          changeset.changes = true;
+          break;
+        }
+      }
+    }
+
+    return changeset;
+  }
+
+  let removedMembersSet = setForArray(oldArray.slice(diff.firstChangeIndex, diff.firstChangeIndex + diff.removedCount));
```

Comment:
Is this actually the set of removed members or rather the range (as a set) that contains all removed members?

Since we scan the range it's not a correctness issue but it makes the code more confusing